ingestor/storage/crds.go

```diff
@@ -0,0 +1,165 @@
+package storage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+
+	adxmonv1 "github.com/Azure/adx-mon/api/v1"
+	"github.com/Azure/adx-mon/pkg/logger"
+	"github.com/Azure/adx-mon/pkg/scheduler"
+	meta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ListFilterFunc func(client.Object) bool
+
+// FilterCompleted will filter out objects that are completed, as determined
+// by their Conditions being in a "True" state and the ObservedGeneration
+// matching the current generation of the object.
+func FilterCompleted(obj client.Object) bool {
+	statusObj, ok := obj.(adxmonv1.ConditionedObject)
+	if !ok {
+		return false
+	}
+
+	condition := statusObj.GetCondition()
+	return condition != nil && condition.Status == metav1.ConditionTrue && condition.ObservedGeneration == obj.GetGeneration()
+}
+
+type CRDHandler interface {
+	List(ctx context.Context, list client.ObjectList, filters ...ListFilterFunc) error
+	UpdateStatus(ctx context.Context, obj client.Object, errStatus error) error
+}
+
+type crdHandler struct {
+	Client  client.Client
+	Elector scheduler.Elector
+}
+
+func NewCRDHandler(client client.Client, elector scheduler.Elector) CRDHandler {
+	return &crdHandler{
+		Client:  client,
+		Elector: elector,
+	}
+}
+
+func (c *crdHandler) List(ctx context.Context, list client.ObjectList, filters ...ListFilterFunc) error {
+	if c.Elector != nil && !c.Elector.IsLeader() {
```

Comment:
I think we'll want to change this (not now) to have the tasks run on all nodes all the time, but partition the tasks to determine which node is the owner and responsible for running the tasks.  Not sure if any of these tasks will incur resource usage but it will let us spread the load more evenly.